### DISCUSSION TITLE
Why media queries are missing from my final file?

### DISCUSSION THREAD

**Discussion by gdarko** [tailwindlabs/tailwindcss#8506] [Help] [answered]
**What version of Tailwind CSS are you using?**

latest

**What build tool (or framework if it abstracts the build tool) are you using?**

i don't use any

**What version of Node.js are you using?**

14.19.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Linux

**Reproduction URL**

https://play.tailwindcss.com/CSO1UPljdM?file=config

**Describe your issue**

The media queries from my file are missing, i can see the responsive screen classes are outside of media queries. I never had this problem on Tailwind 2, but with version 3 many things are broken.

**Comment by adamwathan** [comment, marked as answer]
Hey, this is because you haven’t safelisted the variants you want to generate. The docs explain all of this:

https://tailwindcss.com/docs/content-configuration#using-regular-expressions

What you’re doing here though is not how Tailwind is intended to be used and not a use case we really care to invest in supporting. You should be setting up your `content` configuration and having Tailwind generate classes based on what you are actually using in your templates.

**Reply by gdarko** [reply]
Thanks.

It would be nice if there will be support for non-jit mode, sometimes the classes aren't directly in our files that tailwind3 scans and are generated by some kind of logic, or stored elsewhere (CMS, etc), so it is difficult to use Tailwind 3.0 in such cases.

We used to define a set of plugins and compile minimal set of Tailwind classes (grid, text align, etc), but with 3.0 it is impossible because it is purely jit based.
